### DISCUSSION TITLE
Highlight plugin breaks additionally CSS-classes for codeblocks

### DISCUSSION THREAD

**Discussion by FlTr** [datenstrom/community#78] [Report a bug] [answered]
Short examples:

---

OK:

``````
```{.left}

// here some fancy code
// ...

``````

---

OK:

``````
```javascript

// here some fancy code
// ...

``````

---

Not OK:

``````
```javascript {.left}

// here some fancy code
// ...

``````

---

Have anyone a solution?

**Comment by markseuffert** [comment]
Looks like [PHP Markdown Extra](https://michelf.ca/projects/php-markdown/extra/#fenced-code-blocks) doesn't support blocks with classes both inside and outside the braces.

**Comment by markseuffert** [comment]
``````
``` {javascript .left}
alert('Hello world!')
```
``````

This should be working now. Thanks for testing and reporting bugs! :smile:

**Comment by FlTr** [comment]
great support!

**Comment by FlTr** [comment]
Sorry, but i think it doesn't work. The CSS-class is added but the highlight-tag is ignored. However now it doesn't break the whole layout.

**Comment by markseuffert** [comment]
Can you do me a favour and try it out on the [Yellow demo](https://datenstrom.se/yellow/demo/)?

**Comment by FlTr** [comment]
Uff, I'm a little bit confused. Yes on demo it seems it works, but all code blocks are surrounded with div-tag, and even there is no highlight-tag, their are highlighted ...

![image](https://cloud.githubusercontent.com/assets/9076372/7980434/b2c5ed7e-0aa7-11e5-9a13-f5f482b54b37.png)

**Comment by FlTr** [comment]
This "auto-wrapping" does not appear on my local machine (latest core and highlight-plugin)

**Comment by markseuffert** [comment]
Can you check the version of core and plugins? The demo uses currently:

Yellow 0.5.19
YellowHighlight  0.5.4
YellowMarkdown 0.5.5

What happens in the latest version: The highlight plugin wraps a code block in `<div>...</div>`, even if it doesn't recognize the language. If you use a code block, one without language or class at all, then Markdown generates `<pre>...</pre>`.  Is that what you expected or should it be changed?

**Comment by FlTr** [comment]
Ok, my fault. I thought I updated it but there are more dependencies between core and plugins. With latest versions it works. Thanks.

``` bash
php yellow.php version
Yellow 0.5.19
YellowCommandline 0.5.3
YellowHighlight 0.5.4
YellowMarkdown 0.5.5
```

Anyway there should be a simpler update procedure.

**Comment by FlTr** [comment]
I think all codeblocks should be handled the same way. So add the additional classes to the `<pre>` or surround all blocks with a `<div>`

**Comment by FlTr** [comment]
An extra wish:
Short descripton of the css-classes from _highlight.css_

**Comment by markseuffert** [comment]
Let me see what I can do. I think you are right, all code blocks should be handled the same way.  However, we are using the Geshi syntax highlighting library in the highlight plugin. HTML and CSS for highlighted code blocks come from the library.

**Comment by markseuffert** [comment]
I think I found something that's more predictable. The highlight plugin is only invoked if there's a language identifier. Then the highlight plugin is responsible for generating HTML and CSS. On the other hand if there's no language identifier, then Markdown generates `<pre>...</pre>` without highlighting.

Here is a code block with highlighting:

``````
```javascript
alert('Hello world!')
```
``````

Here are code blocks without highlighting:

``````
```
alert('Hello world!')
```
``````

.

``````
```{.left}
alert('Hello world!') 
```
``````

Hope this works you.

**Comment by markseuffert** [comment]
Waiting for bug being fixed in michelf/php-markdown/pull/200

**Comment by markseuffert** [comment, marked as answer]
There's a new Markdown plugin, syntax highlighting supports optional CSS classes. 

``````
```javascript {.left}
alert('Hello world!')
```
``````

This should be working now.
